Product Component

Chandrika-Rani-Infrrd opened the pull request:
updated:products component with product data containing name and price
updated:products spec file with few test cases
updated:set default path to product component in router module

Chandrika-Rani-Infrrd committed:
Product Component
updated:products component with product data containing name and price
updated:products spec file with few test cases
updated:set default path to product component in router module

lloydaf commented on src/app/products/products.component.spec.ts:
```diff
@@ -22,4 +23,20 @@ describe('ProductsComponent', () => {
   });
+
+  it("does products name match to its price",()=>{
+    const [first,second]=component.products;
```
hmm how does this test make sense?

Chandrika-Rani-Infrrd replied:
different products have different price .if shoes price is 8000 then if i assign any different price value it shows error

Chandrika-Rani-Infrrd replied:
first index should contain the first hard coded name and its value

lloydaf replied:
So you shouldn't write tests for configurations. Configurations can change. Your products can change. It shouldn't matter. Write tests to ensure your code works the way you want it to. :)

lloydaf commented on src/app/products/products.component.spec.ts:
```diff
@@ -22,4 +23,20 @@ describe('ProductsComponent', () => {
+    let button =  fixture.debugElement.query(By.css('button'));
+    button.triggerEventHandler('click',null);
+    fixture.detectChanges();
+    expect(component.onClick).toHaveBeenCalled();
```
💯

lloydaf commented on src/app/products/products.component.html:
```diff
@@ -1 +1,11 @@
+<h1 class="products">Products</h1>
+<div class="productsContainer">
+    <ul class="productList" *ngFor="let product of products">
+        <li class="productContent">{{product.name}}</li>
```
ideally, product should be it's own component

Chandrika-Rani-Infrrd committed:
Product Component
updated:products css file with naming conventions
removed:spec test for configuration
